operator/Makefile

```diff
@@ -103,7 +103,7 @@ help: ## Display this help.
 
 .PHONY: deps
 deps: go.mod go.sum
-	go mod tidy
+	go mod tidy -compat=1.17
```

Comment:
I have a really bad feeling about the error message that `go mod tidy` is outputting when used without the compat option. Maybe we need to discuss splitting out the API dependency as a separate module from Loki?

Reply:
The problem here is our new dependency against Loki itself, in particular we reference: 

```
github.com/grafana/loki/operator/internal/manifests/internal/rules imports
        gopkg.in/yaml.v2 tested by
        gopkg.in/yaml.v2.test imports
        gopkg.in/check.v1 loaded from gopkg.in/check.v1@v1.0.0-20200902074654-038fdea0a05b,
        but go 1.16 would select v1.0.0-20201130134442-10cb98267c6c
github.com/grafana/loki/operator/api/v1beta1 imports
        github.com/grafana/loki/pkg/logql/syntax imports
        github.com/grafana/loki/pkg/util imports
        github.com/grafana/dskit/kv imports
        github.com/grafana/dskit/kv/consul imports
        github.com/hashicorp/consul/api loaded from github.com/hashicorp/consul/api@v1.12.0,
        but go 1.16 would fail to locate it:
        ambiguous import: found package github.com/hashicorp/consul/api in multiple modules:
        github.com/hashicorp/consul v1.2.1 (/home/periklis/.gvm/pkgsets/go1.17.8/global/pkg/mod/github.com/hashicorp/consul@v1.2.1/api)
        github.com/hashicorp/consul/api v1.12.0 (/home/periklis/.gvm/pkgsets/go1.17.8/global/pkg/mod/github.com/hashicorp/consul/api@v1.12.0)

To upgrade to the versions selected by go 1.16, leaving some packages unresolved:
        go mod tidy -e -go=1.16 && go mod tidy -e -go=1.17
If reproducibility with go 1.16 is not needed:
        go mod tidy -compat=1.17
For other options, see:
        https://golang.org/doc/modules/pruning
make: *** [Makefile:106: deps] Error 1
```

Reply:
Requires a patch into loki go.mod to resolve this issue.